roscala/src/main/scala/coop/rchain/rosette/prim/tuple.scala

```diff
@@ -153,6 +153,77 @@ object tuple {
     }
   }
 
+  /**
+    * Define the tuple-head primitive.
+    * This returns the first element of a Tuple
+    * e.g. (tuple-head [1 2 3 4 5 6]) ==> 1
+    */
+  object tplHead extends Prim {
+    override val name: String = "tuple-head"
+    override val minArgs: Int = 1
+    override val maxArgs: Int = 1
+
+    @checkTypeMismatch[Tuple] // Only arg must be a Tuple
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Ob] = {
+      val elem = ctxt.argvec.elem
+
+      checkTuple(0, elem).map { t => // Ensure arg0 is a Tuple
+
+        t.nth(0) match {
+          case Some(v: Ob) => v
+          case None        => Ob.NIV
+        }
+      }
+    }
+  }
+
+  /**
+    * Define the tuple-last primitive.
+    * This returns the last element of a Tuple
+    * e.g. (tuple-last [1 2 3 4 5 6]) ==> 6
+    */
+  object tplLast extends Prim {
+    override val name: String = "tuple-last"
+    override val minArgs: Int = 1
+    override val maxArgs: Int = 1
+
+    @checkTypeMismatch[Tuple] // Only arg must be a Tuple
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Ob] = {
+      val elem = ctxt.argvec.elem
+
+      checkTuple(0, elem).map { t => // Ensure arg0 is a Tuple
+
+        t.nth(t.elem.size - 1) match {
+          case Some(v: Ob) => v
+          case None        => Ob.NIV
+        }
+      }
+    }
+  }
+
+  /**
+    * Define the tuple-tail primitive.
+    * This returns the Tuple consisting of the 2nd through the nth
+    * elements of the provided Tuple
+    * e.g. (tuple-tail [1 2 3 4 5 6]) ==> [2 3 4 5 6]
+    */
+  object tplTail extends Prim {
+    override val name: String = "tuple-tail"
+    override val minArgs: Int = 1
+    override val maxArgs: Int = 1
+
+    @checkTypeMismatch[Tuple] // Only arg must be a Tuple
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Tuple] = {
```

Comment:
What if the `Tuple` is empty? Rosette returns `NIL` in that case.

Reply:
Doesn't it return an empty Tuple?

rosette> (tuple-tail [])
[]
rosette> (tuple-tail[1])
[]

Reply:
`return (n > 0 ? t->makeSlice(1, n-1) : NIL);`

And you see that output because `NIL` is an empty `Tuple`.

Reply:
Done